seacatauth/authn/webauthn/service.py

```diff
@@ -42,20 +48,69 @@ def __init__(self, app, service_name="seacatauth.WebAuthnService"):
 		if self.RelyingPartyId is None:
 			self.RelyingPartyId = str(urllib.parse.urlparse(self.Origin).hostname)
 
+		self.AttestationPreference = asab.Config.get("seacatauth:webauthn", "attestation", fallback="none")
+		# Use "indirect" or "direct" attestation to get metadata about registered authenticators.
+		# Doing so may however impose higher trust requirements on your environment though.
+		if self.AttestationPreference not in frozenset(["none", "direct", "indirect", "enterprise"]):
+			raise ValueError("Unsupported WebAuthn 'attestation' value: {!r}".format(self.AttestationPreference))
+
 		self.RegistrationTimeout = asab.Config.getseconds("seacatauth:webauthn", "challenge_timeout") * 1000
 		self.SupportedAlgorithms = [
 			webauthn.helpers.structs.COSEAlgorithmIdentifier(-7)  # Es256
 		]
 
-		self.KeyNameRegex = re.compile(r"^[a-z][a-z0-9._-]{0,128}[a-z0-9]$")
+		self._FidoMetadataByAAGUID: typing.Dict[int, dict] | None = None
 
 		app.PubSub.subscribe("Application.housekeeping!", self._on_housekeeping)
 
 
+	async def initialize(self, app):
+		await self._load_fido_metadata()
+
+
 	async def _on_housekeeping(self, event_name):
 		await self._delete_expired_challenges()
 
 
+	async def _load_fido_metadata(self, speculative=True):
```

Comment:
- save the metadata in mongo collection
- load on startup if collection empty
- refresh on housekeeping

- alternatively save the JWT in /tmp and store the ETag

Reply:
there is no ETag available